src/main/java/com/rapid7/client/dcerpc/msvcctl/ServiceControlManagerService.java

```diff
@@ -71,41 +71,43 @@ public boolean startService(String service, ContextHandle serviceManagerHandle)
         throws IOException {
         ContextHandle serviceHandle = getServiceHandle(service, serviceManagerHandle);
         RQueryServiceStatusRequest request = new RQueryServiceStatusRequest(serviceHandle);
         RQueryServiceStatusResponse response = transport.call(request);
         if (response.getReturnCode().is(SystemErrorCode.ERROR_SUCCESS.getErrorCode())){
-            return response.getrQueryServiceStatusInfo();
+            return response.getServiceStatusInfo();
         } else return null;
     }
 
-    public RQueryServiceStatusInfo stopService(String service, ContextHandle serviceManagerHandle)
+    public ServiceStatusInfo stopService(String service, ContextHandle serviceManagerHandle)
         throws IOException {
         ContextHandle serviceHandle = getServiceHandle(service, serviceManagerHandle);
         RControlServiceRequest request = new RControlServiceRequest(serviceHandle, RControlServiceRequest.SERVICE_CONTROL_STOP);
         RQueryServiceStatusResponse response = transport.call(request);
         if (response.getReturnCode().is(SystemErrorCode.ERROR_SUCCESS.getErrorCode())){
-            return response.getrQueryServiceStatusInfo();
+            return response.getServiceStatusInfo();
         } else return null;
     }
 
-    public boolean changeServiceConfig(String service, ContextHandle serviceManagerHandle, int serviceType, int startType, int errorControl)
+    public boolean changeServiceConfig(String service,
+                                       ContextHandle serviceManagerHandle,
+                                       ServiceConfigInfo serviceConfigInfo)
         throws IOException {
         ContextHandle serviceHandle = getServiceHandle(service, serviceManagerHandle);
-        RChangeServiceConfigWRequest request = new RChangeServiceConfigWRequest(serviceHandle, serviceType, startType, errorControl);
+        RChangeServiceConfigWRequest request = new RChangeServiceConfigWRequest(serviceHandle, serviceConfigInfo);
         RChangeServiceConfigWResponse response = transport.call(request);
         return (response.getReturnCode().is(SystemErrorCode.ERROR_SUCCESS.getErrorCode()));
 
     }
 
-    public RQueryServiceConfigInfo queryServiceConfig(String service, ContextHandle serviceManagerHandle) throws IOException {
+    public ServiceConfigInfo queryServiceConfig(String service, ContextHandle serviceManagerHandle) throws IOException {
         ContextHandle serviceHandle = getServiceHandle(service, serviceManagerHandle);
         RQueryServiceConfigWRequest request = new RQueryServiceConfigWRequest(serviceHandle, RQueryServiceConfigWRequest.MAX_BUFFER_SIZE);
         RQueryServiceConfigWResponse response = transport.call(request);
         if (response.getReturnCode().is(SystemErrorCode.ERROR_SUCCESS.getErrorCode())){
-            return response.getrQueryServiceConfigInfo();
+            return response.getServiceConfigInfo();
```

Comment:
We should throw exception that contains the error code so that caller can handle the failure.